service/history/workflow/mutable_state.go

```diff
@@ -362,6 +364,8 @@ type (
 		)
 		UpdateActivity(*persistencespb.ActivityInfo) error
 		UpdateActivityWithTimerHeartbeat(*persistencespb.ActivityInfo, time.Time) error
+		UpdateActivityWithCallback(*persistencespb.ActivityInfo, UpdateActivityCallback) error
+		UpdatePauseInfoSearchAttribute() error
```

Comment:
Don't export IMO, this can be an implementation detail.

Reply:
yes, with the new changes this one shouldn't be expoerted.